app/domain/authentication/util/check_authenticator_secret_exists.rb

```diff
@@ -0,0 +1,40 @@
+require 'command_class'
+require 'conjur/fetch_required_secrets'
+
+module Authentication
+  module Util
+
+    # Command class that returns boolean value indicating if resource exists
+    CheckAuthenticatorSecretExists = CommandClass.new(
```

Comment:
since this is a generic command class you should handle an authenticator which not has service id( like gcp)

Reply:
but fetch_authenticator_secrets also recieves service_id and its already in the code ?

Reply:
it is not a blocker and should be in different ticket but currently we don't support authentications without service_id